AIAgent/common/config/validation_config.py

```diff
@@ -0,0 +1,36 @@
+from abc import ABC
+from typing import Literal, Union
+
+from common.config.platform_config import Platform
+from pydantic import Field
+from pydantic.dataclasses import dataclass as pydantic_dataclass
+
+
+@pydantic_dataclass
+class Validation(ABC):
+    val_type: str
+
+
+@pydantic_dataclass
+class ValidationLoss(Validation):
+    val_type: Literal["loss"]
+    batch_size: int
+
+
+@pydantic_dataclass
+class ValidationSVM(Validation, ABC):
+    platforms_config: list[Platform] = Field(alias="PlatformsConfig")
+    fail_immediately: bool = Field(default=False)
+
+
+@pydantic_dataclass
+class ValidationSVMViaServer(ValidationSVM):
+    val_type: Literal["svms_server"]
+    servers_count: int = Field()  # type: ignore
```

Comment:
```suggestion
    servers_count: int
```
?

Reply:
TypeError: non-default argument 'servers_count' follows default argument

Reply:
`val_type: Literal["svms_server"]` is most likely a redefinition of `super().val_type`, because it is declared again in child class

please check the comment below, see if it works

Reply:
`val_type` is not defined in either Validation or ValidationSVM. Or what do you mean? 🙂

Reply:
I see, `ValidationConfig` is used as a mode switch in `run_training`: https://github.com/PySymGym/PySymGym/blob/d9099aae56e29dd57b30b182a29ad80809b9937c/AIAgent/run_training.py#L106

this is very sus, but I see no low-effort solution so I guess it stays